### PR TITLE
Migrate onnxrt RTN WOQ to 3.x API

### PR DESCRIPTION
**Pull request by yuwenzho**
## Type of Change

feature
API changed or not: no

## Description

Migrate onnxrt RTN WOQ to 3.x API

## How has this PR been tested?

CI

## Dependency Change?

No

### CODE REVIEW

File: neural_compressor/onnxrt/algorithms/weight_only/rtn.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+
+import numpy as np
```
Comment (yiliu30) [inline neural_compressor/onnxrt/algorithms/weight_only/rtn.py:24]: Add `numpy` to requirements_onnxrt.txt?

**Review by yiliu30 — APPROVED** [review]

File: neural_compressor/onnxrt/quantization/config.py [hunk 2]
```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023 Intel Corporation
```
[the remark this answers is not in the document]
Comment (chensuyue) [reply neural_compressor/onnxrt/quantization/config.py:4]: Yes, we should.

[the remark this answers is not in the document]
Comment (yiliu30) [reply neural_compressor/onnxrt/quantization/config.py:4]: We'd better update the license template utilized by pre-commit to address this common issue.

File: neural_compressor/onnxrt/algorithms/weight_only/rtn.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+        # check op_type of node is MatMul
+        # check dim 1 of input is weight tensor
+        # check node config is RTNWeightQuantConfig
+        # check node weight_dtype config is not fp32
+        if (
+            node.op_type in ["MatMul"]  # check op_type of node is MatMul
+            and model.get_initializer(node.input[1]) is not None
+            and isinstance(weight_config.get((node.name, node.op_type), {}), RTNWeightQuantConfig)
+            and weight_config.get((node.name, node.op_type), {}).weight_dtype.lower() != "fp32"
```
Comment (mengniwang95) [inline neural_compressor/onnxrt/algorithms/weight_only/rtn.py:190]: if weight_config is {}, '.weight_dtype' will raise error

Comment (yuwenzho) [reply neural_compressor/onnxrt/algorithms/weight_only/rtn.py:190]: if weight_config is {}, `isinstance(weight_config.get((node.name, node.op_type), {}), RTNWeightQuantConfig)` will be False and L190 will not be executed

File: neural_compressor/onnxrt/algorithms/weight_only/algo_entry.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+from neural_compressor.onnxrt.utils.utility import register_algo
+
+logger = Logger().get_logger()
+
+
+###################### RTN Algo Entry ##################################
+@register_algo(name=RTN_WEIGHT_ONLY_QUANT)
+def rtn_quantize_entry(
+    model: Union[Path, str],
+    configs_mapping: Dict[Tuple[str, callable], RTNWeightQuantConfig],
```
Comment (mengniwang95) [inline neural_compressor/onnxrt/algorithms/weight_only/algo_entry.py:33]: As we discussed, expand quant_config in algo_entry.py and replace this param name with quant_config

File: neural_compressor/onnxrt/quantization/quantize.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+
+def need_apply(configs_mapping: Dict[Tuple[str, callable], BaseConfig], algo_name):
+    return any(config.name == algo_name for config in configs_mapping.values() if hasattr(config, "name"))
+
+
+# only for internal usage now
+def _quantize(
+    model_input: Tuple[Path, str],
+    quant_config: BaseConfig,
```
Comment (mengniwang95) [inline neural_compressor/onnxrt/quantization/quantize.py:35]: need calibrator param, we can use calibration_data_reader to align with ORT

File: neural_compressor/onnxrt/quantization/quantize.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+from neural_compressor.common.base_config import BaseConfig, ComposableConfig, config_registry
+from neural_compressor.common.logger import Logger
+from neural_compressor.onnxrt.quantization.config import FRAMEWORK_NAME
+from neural_compressor.onnxrt.utils.utility import algos_mapping
+
+logger = Logger().get_logger()
+
+
+def need_apply(configs_mapping: Dict[Tuple[str, callable], BaseConfig], algo_name):
+    return any(config.name == algo_name for config in configs_mapping.values() if hasattr(config, "name"))
```
Comment (mengniwang95) [inline neural_compressor/onnxrt/quantization/quantize.py:29]: if we expand config in algo_entry, here should be quant_config.name == algo_name

File: test/3x/onnxrt/quantization/weight_only/test_rtn.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+
+class TestRTNQuant(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        main_export(
+            "hf-internal-testing/tiny-random-gptj",
+            output="gptj",
+        )
+        self.gptj = "gptj/model.onnx"
```
Comment (mengniwang95) [inline test/3x/onnxrt/quantization/weight_only/test_rtn.py:18]: optimum >= 1.14.0?

**Review by mengniwang95 — APPROVED** [review]

**Pull request merged**